contributors/design-proposals/customresources-validation.md

```diff
@@ -0,0 +1,525 @@
+# Validation for CustomResources
+
+Authors: @nikhita, @sttts, some ideas integrated from @xiao-zhou’s proposal<sup id="f1">[1](#footnote1)</sup>
+
+
+## Table of Contents
+
+1. [Overview](#overview)
+2. [Background](#background)
+    1. [Goals](#goals)
+    2. [Non-Goals](#non-goals)
+3. [Proposed Extension of CustomResourceDefinition](#proposed-extension-of-customresourcedefinition)
+    1. [API Types](#api-types)
+    2. [Examples](#examples)
+        1. [JSON-Schema](#json-schema)
+        2. [Error messages](#error-messages)
+4. [Validation Behavior](#validation-behavior)
+    1. [Metadata](#metadata)
+    2. [Server-Side Validation](#server-side-validation)
+    3. [Client-Side Validation](#client-side-validation)
+    4. [Comparison between server-side and client-side Validation](#comparison-between-server-side-and-client-side-Validation)
+    5. [Existing Instances and changing the Schema](#existing-instances-and-changing-the-schema)
+    6. [Outlook to Status Sub-Resources](#outlook-to-status-sub-resources)
+    7. [Outlook Admission Webhook](#outlook-admission-webhook)
+5. [Implementation Plan](#implementation-plan)
+6. [Appendix](#appendix)
+    1. [Expressiveness of JSON-Schema](#expressiveness-of-json-schema)
+    2. [JSON-Schema Validation Runtime Complexity](#json-schema-validation-runtime-complexity)
+    3. [Alternatives](#alternatives)
+        1. [Direct Embedding of the Schema into the Spec](#direct-embedding-of-the-schema-into-the-spec)
+        2. [External CustomResourceSchema Type](#external-customresourceschema-type)
+
+## Overview
+
+This document proposes the design and describes a way to add JSON-Schema based validation for Custom Resources.
+
+## Background
+
+ThirdPartyResource (TPR) is deprecated and CustomResourceDefinition (CRD) is the successor which solves the fundamental issues of TPRs to form a stable base for further features.
```

Comment:
Is it outside the scope of this doc to briefly mention what those issues are? Perhaps just as links